iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/CatalogWrapperForREST.java

```diff
@@ -0,0 +1,157 @@
+    super(config);
+    this.catalogConfigToClients =
+        MapUtils.getFilteredMap(
+            config.getIcebergCatalogProperties(),
+            key -> catalogPropertiesToClientKeys.contains(key));
+    // To compatibility with old version
+    Map<String, String> catalogProperties = normalizeCredentialProperties(config.getAllConfig());
+    this.catalogCredentialManager = new CatalogCredentialManager(catalogName, catalogProperties);
+  }
+
+  public LoadTableResponse createTable(
+      Namespace namespace, CreateTableRequest request, boolean requestCredential) {
+    LoadTableResponse loadTableResponse = super.createTable(namespace, request);
+    if (requestCredential) {
+      return injectCredentialConfig(
+          TableIdentifier.of(namespace, request.name()), loadTableResponse);
+    }
+    return loadTableResponse;
+  }
+
+  public LoadTableResponse loadTable(TableIdentifier identifier, boolean requestCredential) {
+    LoadTableResponse loadTableResponse = super.loadTable(identifier);
+    if (requestCredential) {
+      return injectCredentialConfig(identifier, loadTableResponse);
+    }
+    return loadTableResponse;
+  }
+
+  @Override
+  public void close() {
+    if (catalogCredentialManager != null) {
+      catalogCredentialManager.close();
+    }
+  }
+
+  private Map<String, String> getCatalogConfigToClient() {
+    return catalogConfigToClients;
+  }
+
+  private LoadTableResponse injectCredentialConfig(
+      TableIdentifier tableIdentifier, LoadTableResponse loadTableResponse) {
+    TableMetadata tableMetadata = loadTableResponse.tableMetadata();
+    String[] path =
+        Stream.of(
+                tableMetadata.location(),
+                tableMetadata.property(TableProperties.WRITE_DATA_LOCATION, ""),
+                tableMetadata.property(TableProperties.WRITE_METADATA_LOCATION, ""))
+            .filter(StringUtils::isNotBlank)
+            .toArray(String[]::new);
+
+    PathBasedCredentialContext context =
+        new PathBasedCredentialContext(
+            PrincipalUtils.getCurrentUserName(), ImmutableSet.copyOf(path), Collections.emptySet());
+    Credential credential = catalogCredentialManager.getCredential(context);
+    if (credential == null) {
+      throw new ServiceUnavailableException("Couldn't generate credential, %s", context);
+    }
+
+    LOG.info(
+        "Generate credential: {} for Iceberg table: {}",
+        credential.credentialType(),
+        tableIdentifier);
+
+    Map<String, String> credentialConfig = CredentialPropertyUtils.toIcebergProperties(credential);
+    return LoadTableResponse.builder()
+        .withTableMetadata(loadTableResponse.tableMetadata())
+        .addAllConfig(loadTableResponse.config())
+        .addAllConfig(getCatalogConfigToClient())
+        .addAllConfig(credentialConfig)
+        .build();
+  }
+
+  @SuppressWarnings("deprecation")
+  private Map<String, String> normalizeCredentialProperties(Map<String, String> properties) {
```

Comment:
I think the name "normalize" cannot fully describe the meaning of this method, this method doesn't normalize anything, it just check and keep the compatibility of the deprecated configs. It would be better to have a new name.